subprojects/core/src/main/java/org/gradle/api/internal/artifacts/configurations/ConfigurationRolesForMigration.java

```diff
@@ -27,7 +27,8 @@
  * from one role to another by marking usages which are present in the current role but not present in the eventual role
  * as deprecated.
  *
- * <p>The roles here are all meant to be temporary roles used for migration only, to be removed in Gradle 9.0.</p>
+ * <p>The remaining role here is meant to be a temporary roles used for migration only, to be removed in Gradle 10.0
```

Comment:
```suggestion
 * <p>
 * The roles here is meant to be a temporary roles used for migration only. 
 * Upon major version releases, configurations in a migration role should be updated 
 * to their destination role.
```

I think this class might maintain some usefulness across versions, and removing version numbers from the javadoc might help reduce churn here.

Some examples of migration roles that could be useful would be:
resolvable dependency scope -> resolvable
resolvable dependency scope -> dependency scope
any role -> none

The `none` migration role being a way for us to deprecate a configuration for _all_ roles, which would be useful if we plan to remove a configuration.